### PR TITLE
Refactor BaseUnitTests and list tests following new test template

### PR DESCRIPTION
**Pull request by arn3tt**

### CODE REVIEW

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 1]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
 
     @Test
     public void testAddFirst() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 2]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
...
         this.list.addItem(order);
 
+        // verify
         assertEquals(order, this.list.getHead().getOrder());
         assertEquals(order, this.list.getCurrent().getOrder());
         assertEquals(order, this.list.getTail().getOrder());
     }
 
     @Test
     public void testAddOrder() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 3]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
...
-        // addItem() should have fixed this.current to point to the newly item added to the tail of
-        // the list.
+        // verify
+        // addItem() should have fixed this.current to point to the newly item added
+        // to the tail of the list.
         assertEquals(orderThree, this.list.getNext());
     }
 
     @Test
     public void testAddNullOrder() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 4]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
...
             this.list.addItem(orderNull);
             fail("Null order should not be added.");
         } catch (IllegalArgumentException e) {
+            // verify
             assertEquals("Attempting to add a null order.", e.getMessage());
         }
     }
 
     @Test
     public void testResetPointer() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 5]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
...
             this.list.removeItem(orderNull);
             fail("Null order should not be removed.");
         } catch (IllegalArgumentException e) {
+            // verify
             assertEquals("Attempting to remove a null order.", e.getMessage());
         }
     }
 
     @Test
     public void testFindNodeToRemove() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 6]
```diff
@@ -126,23 +160,31 @@ public void testFindNodeToRemove() {
 
     @Test
     public void testRemoveItemOnHead() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 7]
```diff
@@ -155,19 +197,27 @@ public void testRemoveItemOnHead() {
 
     @Test
     public void testRemoveItemOnTail() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 8]
```diff
@@ -177,15 +227,24 @@ public void testRemoveItemOnTail() {
 
     @Test
     public void testRemoveItemOneElementOnList() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 9]
```diff
@@ -225,28 +291,40 @@ public void testRemoveItem() throws Exception {
 
     @Test
     public void testReinitializingList() {
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: The test cases must be described before testing based on the new template.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java [hunk 10]
```diff
@@ -24,99 +24,133 @@ public void initialize() {
 
     @Test
     public void testAddFirst() {
+        // verify
         assertNull(this.list.getHead());
         assertNull(this.list.getCurrent());
         assertNull(this.list.getTail());
```
Comment (wassoares) [inline src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: Although it is not in the template, it was defined that all Assert must also not be static.

Comment (arn3tt) [reply src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: Nice catch, will fix this.

File: src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java (hunk 1 above)
Comment (arn3tt) [reply src/test/java/org/fogbowcloud/manager/core/models/linkedlists/SynchronizedDoublyLinkedListTest.java]: Right! I will fix this in the next commit.

**Pull request merged**